projects/ngx-skeleton-loader/src/lib/directives/skeleton-loader.directive.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  ComponentFactoryResolver,
+  Directive,
+  ElementRef,
+  Inject,
+  Input,
+  OnChanges, Optional,
+  ViewContainerRef
+} from '@angular/core';
+import { NGX_SKELETON_LOADER_CONFIG, NgxSkeletonLoaderConfig, } from "../ngx-skeleton-loader-config.types";
+import { NgxSkeletonLoaderComponent } from "../ngx-skeleton-loader.component";
+
+@Directive({
+  selector: '[ngxSkeletonShow], [ngxSkeletonStyle], [ngxSkeletonCount], [ngxSkeletonType], [ngxSkeletonUseParent]',
+})
+export class SkeletonLoaderDirective implements OnChanges {
```

Comment:
This directive is heavily checking for different things: CSS Styling, Repaints, reflows and such. I would challenge some approaches here:

- Why not adding CSS classes  instead of having some hardcoded numbers inside the directive?
- There are few literals here (E.G. 5, 10, 100)? My concern with that is cross browser requirements;
- Few `any` types here that should be avoided for something less permissive;
- Last, but not least: it needs unit tests for all these new cases;

Reply:
Interesting findings 👍.. heavily repaints? Just curious how did you confirm that? It would be helpful for me to fix that problem.

>  Why not adding CSS classes instead of having some hardcoded numbers inside the directive?
> There are few literals here (E.G. 5, 10, 100)? My concern with that is cross browser requirements;

Hmm.. I don't think you can use CSS classes here. Unless I am missing something.  Are you able to give me an example? Those default dimensions would be used only when the computed dimension is zero and when the users do not pass in the dimensions.  

> Few any types here that should be avoided for something less permissive;
> Last, but not least: it needs unit tests for all these new cases;
Good catch. I can certainly fix the any and add some unit test for these.